### PR TITLE
WFLY-14877 Replace jboss.web.allow-orphan-session system property with formal Undertow servlet-container attribute

### PR DESCRIPTION
**Pull request by pferraro**
Completes https://issues.redhat.com/browse/WFLY-14877, replacing the system property with a formal Undertow servlet-container attribute.
Drops UNDERTOW-1902 workaround.

Consolidates all parser classes into a single parameterized PersistentResourceXMLDescription factory.
Includes necessary transformers and transformer tests.

### CODE REVIEW

**Comment by rhusar** [comment]
@pferraro BTW this needs a rebase now.

**Review by fl4via — APPROVED** [review]

File: undertow/src/main/java/org/wildfly/extension/undertow/UndertowPersistentResourceXMLDescriptionFactory.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+        Stream<AttributeDefinition> ssoAttributes = Stream.concat(Attribute.stream(ApplicationSecurityDomainSingleSignOnDefinition.Attribute.class), Attribute.stream(SingleSignOnDefinition.Attribute.class));
+        builder.addChild(builder(SingleSignOnDefinition.PATH_ELEMENT, ssoAttributes));
+
+        Stream<AttributeDefinition> attributes = ApplicationSecurityDomainDefinition.ATTRIBUTES.stream();
+        attributes.forEach(builder::addAttribute);
+        return builder;
+    }
+
+    // TODO Drop methods below once WFCORE-6218 is available
```
Comment (fl4via) [inline undertow/src/main/java/org/wildfly/extension/undertow/UndertowPersistentResourceXMLDescriptionFactory.java:274]: I see that the PR for WFCORE-6218 is already reviewed: https://github.com/wildfly/wildfly-core/pull/5369 so hopefully we will be able to drop these pretty soon. Should I create a follow up Jira, or should we assign this to WFLY-14877?

Comment (pferraro) [reply undertow/src/main/java/org/wildfly/extension/undertow/UndertowPersistentResourceXMLDescriptionFactory.java:274]: I created https://issues.redhat.com/browse/WFLY-17611 to track this.  I can handle that cleanup once the requisite wildfly-core version is upgraded.

**Pull request merged**